autoload/rust.vim

```diff
@@ -208,6 +280,23 @@ function! rust#AppendCmdLine(text)
 	return cmd
 endfunction
 
+" Tokenize the String according to shell parsing rules
+function! s:ShellTokenize(text)
```

Comment:
Good point, I only considered sh-compatible shells (as that implements the sh tokenizing rules). Perhaps the best bet would be to also remove all the escape characters during tokenizing, so it can be passed back to `shellescape()`. It could theoretically implement platform-native tokenizing rules but I don't have a clue what the Windows shell rules are, and personally, I don't see a problem with just defining it as the sh rules.